app/src/main/java/com/codesoom/assignment/handler/DemoHttpHandler.java

```diff
@@ -0,0 +1,286 @@
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DemoHttpHandler implements HttpHandler {
+
+    private static Long taskIdSeq = 0L;
+
+    private List<Task> tasks = new ArrayList<>();
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+
+        String method = httpExchange.getRequestMethod(); // Method
+        URI uri = httpExchange.getRequestURI(); // URI
+        String path = uri.getPath(); // Path
+        Long pathValue = null; // PathValue로 넘겨받은 값
+
+        System.out.println(method + " " + path);
+
+        try {
+
+            Task rqBody = new Task(); // Request에서 받은 데이터를 담을 객체
+
+            // 입력값 가져오기
+            InputStream inputStream = httpExchange.getRequestBody();
+            String body = new BufferedReader(new InputStreamReader(inputStream))
+                    .lines()
+                    .collect(Collectors.joining("\n"));
+
+            if(!body.isBlank()){ // Request에서 받은 데이터가 있다면 Task 객체에 넣음
+                rqBody = JsonUtil.jsonStrToTask(body);
+            }
+
+            // PathVariable이 있는지 검사한다.
+            String[] splitArr = path.split("/");
+            if(splitArr.length > 2){  // PathValue가 있을 경우 값 세팅 및 PathValue에 해당하는 Task의 인덱스를 구함
+
+                // 값 세팅
+                path = "/" + splitArr[1]; // API 경로
+                pathValue = Long.valueOf(splitArr[2]); // PathValue(taskId)
+
+            }
+
+            if(method.equals(HttpMethod.GET.name()) && path.equals("/tasks")){ // Task List 조회 & 단건조회
+
+                if(pathValue == null){
+                    tasksGET(httpExchange);
+                } else {
+                    tasksGET(httpExchange, pathValue);
+                }
+
+
+            } else if(method.equals(HttpMethod.POST.name()) && path.equals("/tasks")){ // Task 등록
+
+                tasksPOST(httpExchange, rqBody);
+
+            } else if( method.equals(HttpMethod.PUT.name()) && path.equals("/tasks") ){ // Task 수정
+
+                tasksPUT(httpExchange, pathValue, rqBody);
+
+            } else if( method.equals(HttpMethod.PATCH.name()) && path.equals("/tasks") ){ // Task 수정
+
+                tasksPATCH(httpExchange, pathValue, rqBody);
+
+            } else if(method.equals(HttpMethod.DELETE.name()) && path.equals("/tasks")){ // Task 삭제
+
+                tasksDELETE(httpExchange, pathValue);
+
+            } else {
+
+                sendResponse(httpExchange, HttpStatus.NOT_FOUND.getCodeNo(), "잘못된 요청입니다.");
+
+            }
+
+        } catch (NotFoundTaskException e) {
+
+            e.printStackTrace();
+            sendResponse(httpExchange, HttpStatus.NOT_FOUND.getCodeNo(), "해당하는 Task를 찾을 수 없습니다.");
+
+        } catch (Exception e) {
+
+            e.printStackTrace();
+            sendResponse(httpExchange, HttpStatus.NOT_FOUND.getCodeNo(), "통신에 실패했습니다.");
+
+        }
+
+    }
+
+    /**
+     * Task List에서 ID에 해당하는 Task의 Index를 찾음

```

Comment:
JavaDoc을 작성하셨군요. 훌륭합니다. 그런데 기왕 작성하는 거 제대로 작성하면 좋겠지요? 

다음은 제가 수정한 JavaDoc summary 입니다. 어떤 점이 다른 것 같나요?
```suggestion
     * 할 일 목록에서 주어진 할 일 id에 해당하는 할 일의 인덱스 넘버를 찾아 리턴합니다.
     * @param tasks 할 일 목록
     * @param taskId 찾으려 하는 할 일의 아이디
```

Reply:
음... 
일단 파라미터가 무엇을 의미하는지 다 적어주셨군요.

주석은 다른 개발자가 혹은 훗날의 자신이 이 소스를 리딩 할 떄 도움이 되도록 적는 용도로 이해하고 있습니다.

그런 주석에서 메서드에 대한 설명을 적을 때 Input과 Output이 무엇인지에 대한 내용을 적어  주는게 당연한데..
단순한 메서드이다 보니 귀찮음에 적지 않고 그냥 넘어간것같습니다.
반성합니다.. 흐..

메서드 설명 또한 파라미터 변수명 그대로 써서 적는 것 보다는
트레이너님이 적어주신 것 처럼 아래 "@param"에 파라미터 변수의 의미로 적어놓은 단어를 사용해서 설명을 적는게 
보는 이로 하여금 의미가 좀 더 명확하게 전달 될 것 같습니다.

이 부분들은 수정해서 다시 Push해 놓았습니다.

다음부턴 간단한 로직이라도 주석을 꼼꼼히 적도록 명심하겠습니다.

감사합니다,

Reply:
앞으로 JavaDoc에 대해서 놀라운 경험을 하실 일들이 많을 거에요 :-)